azkaban-common/src/main/java/azkaban/AzkabanCommonModule.java

```diff
@@ -21,54 +21,40 @@
 import azkaban.spi.Storage;
 import azkaban.spi.StorageException;
 import azkaban.storage.LocalStorage;
-import azkaban.storage.StorageConfig;
 import azkaban.storage.StorageImplementationType;
 import azkaban.utils.Props;
 import com.google.inject.AbstractModule;
 import com.google.inject.Inject;
 import com.google.inject.Provides;
 import com.google.inject.Scopes;
-import com.google.inject.Singleton;
 import java.io.File;
 
-import static azkaban.storage.StorageImplementationType.*;
 
 /**
  * This Guice module is currently a one place container for all bindings in the current module. This is intended to
  * help during the migration process to Guice. Once this class starts growing we can move towards more modular
  * structuring of Guice components.
  */
 public class AzkabanCommonModule extends AbstractModule {
-  private final Props props;
-  /**
-   * Storage Implementation
-   * This can be any of the {@link StorageImplementationType} values in which case {@link StorageFactory} will create
-   * the appropriate storage instance. Or one can feed in a custom implementation class using the full qualified
-   * path required by a classloader.
-   *
-   * examples: LOCAL, DATABASE, azkaban.storage.MyFavStorage
-   *
-   */
-  private final String storageImplementation;
+  private final AzkabanCommonModuleConfig config;
 
   public AzkabanCommonModule(Props props) {
-    this.props = props;
-    this.storageImplementation = props.getString(Constants.ConfigurationKeys.AZKABAN_STORAGE_TYPE, DATABASE.name());
+    this.config = new AzkabanCommonModuleConfig(props);
   }
 
   @Override
   protected void configure() {
     bind(ProjectLoader.class).to(JdbcProjectLoader.class).in(Scopes.SINGLETON);
-    bind(Props.class).toInstance(props);
+    bind(Props.class).toInstance(config.getProps());
     bind(Storage.class).to(resolveStorageClassType()).in(Scopes.SINGLETON);
   }
 
   public Class<? extends Storage> resolveStorageClassType() {
```

Comment:
I was thinking of it initially. But I also don't want too much business logic inside a config class. I'm leaving class loader specific code inside the module because that's what a module should do.